src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -640,23 +640,27 @@ ValidatorList::updateTrusted(hash_set<NodeID> const& seenValidators)
     }
 
     TrustChanges trustChanges;
-    {
-        hash_set<PublicKey> newTrustedKeys;
 
-        for (auto const& val : keyListings_)
+    auto it = trustedKeys_.cbegin();
+    while (it != trustedKeys_.cend())
+    {
+        if (! keyListings_.count(*it) ||
+            validatorManifests_.revoked(*it))
         {
-            if (validatorManifests_.revoked (val.first))
-                continue;
-
-            newTrustedKeys.insert(val.first);
-            if (trustedKeys_.erase(val.first) == 0)
-                trustChanges.added.insert(calcNodeID(val.first));
+            trustChanges.removed.insert(calcNodeID(*it));
+            it = trustedKeys_.erase(it);
         }
+        else
+        {
+            ++it;
+        }
```

Comment:
Shouldn't the iterator be incremented in all cases? Otherwise, there's an extra iteration through the loop to check whether it's there or not?

Reply:
It get "incremented" to the iterator returned by `erase`
https://github.com/ripple/rippled/pull/2613/files/5f549e60118619774201b68fdcf90f4cccc69c9b..b66aa3810f50150e0dc13afa36b74f7237bc5001#diff-7f8c7a926e17debbef064bc5c2f572dbR651

Reply:
Now I see.
http://www.cplusplus.com/reference/map/map/erase/
"The other versions return an iterator to the element that follows the last element removed (or map::end, if the last element was removed)."